lib/context.js

```diff
@@ -117,8 +117,26 @@ var proto = module.exports = {
       return;
     }
 
-    // unset all headers
+    // unset all headers not explicitly kept
+    // while it's inconvenient to use setHeader and getHeader,
+    // res._headers isn't documentented and can't be relied on much
+    // for example: the keys are always stored in lowercase
+    var errHeaders = err.headers ? Object.keys(err.headers) : [];
+    var newHeaders = {};
+    for (var i = 0; i < errHeaders.length; i++) {
```

Comment:
`for (var key in err.headers)` is ok and have the similar performance with a small number of key.